frontend/public/components/catalog-items.jsx

```diff
@@ -5,37 +5,126 @@ import {Link} from 'react-router-dom';
 import {CatalogTileView} from 'patternfly-react-extensions/dist/esm/components/CatalogTileView';
 import {CatalogTile} from 'patternfly-react-extensions/dist/esm/components/CatalogTile';
 import {VerticalTabs} from 'patternfly-react-extensions/dist/esm/components/VerticalTabs';
+import {FilterSidePanel} from 'patternfly-react-extensions/dist/esm/components/FilterSidePanel';
+import {EmptyState} from 'patternfly-react/dist/esm/components/EmptyState';
+import FormControl from 'patternfly-react/dist/esm/components/Form/FormControl';
 
 import {normalizeIconClass} from './catalog-item-icon';
-import {categorizeItems} from './utils/categorize-catalog-items';
+import {categorizeItems, reCategorizeItems} from './utils/categorize-catalog-items';
 
 export class CatalogTileViewPage extends React.Component {
   constructor(props) {
     super(props);
 
-    this.state = {
-      showAllItemsForCategory: null,
-      activeTabs: ['all'], // array of tabs [main category, sub-category]
+    const allCategories = categorizeItems(props.items);
+    this.initialCategories = _.cloneDeep(allCategories);
```

Comment:
This should be in component state.

Is this the categories before filtering is applied? If so, I'd call this `unfilteredCategories`.